ctdcal/process_ctd.py

```diff
@@ -1786,45 +1875,104 @@ def export_bin_data(df, ssscc, sample_rate, search_time, p_column_names, p_col='
+                    print(col + " missing, flagging with 9s")
+                    df[col] = 9
+                else:
+                    print(col + " missing, filling with -999s")
+                    df[col] = -999
+
+    # load other necessary params
+    # TODO: this will change when config.py improves, find different method
+    expocode = cfg.cruise["expocode"]
+    section_id = cfg.cruise["sectionid"]
+    ctd = cfg.ctd_serial
+    out_dir = 'data/pressure/'
+    p_col = 'CTDPRS'
+    stacst_col = 'SSSCC'
+    logFile = 'data/logs/cast_details.csv'
 
     df[stacst_col] = df[stacst_col].astype(str).copy()
 
-    cast_details = dataToNDarray(logFile,str,None,',',0)
-    depth_df = pd.read_csv('data/logs/depth_log.csv')
-    depth_df.dropna(inplace=True)
+    cast_details = pd.read_csv("data/logs/cast_details.csv", dtype={"SSSCC": str})
+    depth_df = pd.read_csv('data/logs/depth_log.csv').dropna()
     manual_depth_df = pd.read_csv('data/logs/manual_depth_log.csv')
-    full_depth_df = pd.concat([depth_df,manual_depth_df])
+    full_depth_df = pd.concat([depth_df,manual_depth_df])  # TODO: update from STNNBR to SSSCC
     full_depth_df.drop_duplicates(subset='STNNBR', keep='first',inplace=True)
 
-    for cast in ssscc:
+    for ssscc in ssscc_list:
 
-        time_data = df[df['SSSCC'] == cast].copy()
+        time_data = df[df['SSSCC'] == ssscc].copy()
+        time_data = pressure_sequence(time_data)
+        time_data = flag_backfill_data(time_data)
+        time_data = fill_surface_data(time_data)
+        time_data = time_data[p_column_names]
+        time_data = time_data.round(4)
 
-        s_num = cast[-5:-2]
-        c_num = cast[-2:]
+        s_num = ssscc[-5:-2]
+        c_num = ssscc[-2:]
         depth = full_depth_df.loc[full_depth_df['STNNBR'] == int(s_num),'DEPTH']
-        for line in cast_details:
-            if cast in line[0]:
-                for val in line:
-                    if 'at_depth' in val: btime = float(str.split(val, ':')[1])
-                    if 'latitude' in val: btm_lat = float(str.split(val, ':')[1])
-                    if 'longitude' in val: btm_lon = float(str.split(val, ':')[1])
-                    if 'altimeter_bottom' in val: btm_alt = float(str.split(val, ':')[1])
-                break
-        time_data.drop(columns='SSSCC',inplace=True)
-        bdt = datetime.datetime.fromtimestamp(btime).strftime('%Y%m%d %H%M').split(" ")
-        b_date = bdt[0]
-        b_time = bdt[1]
-        now = datetime.datetime.now()
+        # get cast_details for current SSSCC
+        cast_dict = cast_details[cast_details["SSSCC"] == ssscc].to_dict("r")[0]
+        b_datetime = datetime.fromtimestamp(cast_dict["bottom_time"], tz=timezone.utc).strftime('%Y%m%d %H%M').split(" ")
+        b_date = b_datetime[0]
+        b_time = b_datetime[1]
+        btm_lat = cast_dict["latitude"]
+        btm_lon = cast_dict["longitude"]
+        btm_alt = cast_dict["altimeter_bottom"]
+
+        now = datetime.now()
```

Comment:
All dates in the library should probably be timezone aware and be set to UTC, even for something trivial like how this will be used:
```suggestion
        now = datetime.now(timezone.utc)
```